lib/open_project/translations/models/i18n_provider.rb

```diff
@@ -85,17 +86,44 @@ def download_locales(output)
     end
   end
 
-  def each_locale
+  def open_locale_zip(force: false)
     begin
-      languages_files = create_temp_file('crowdin_translations')
-      download_locales(languages_files.path)
-      Zip::File.open(languages_files.path) do |zip_file|
-        zip_file.glob("*/#{@crowdin_directory}/*.yml").each do |entry|
-          yield entry
-        end
+      # download if not downloaded or downloaded for wrong project
+      if defined? @languages_files == nil \
+        || @languages_files.length == 0 \
+        || wrong_languages_files?
+        @languages_files = create_temp_file('crowdin_translations')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.